src/Node/Expression/ListExpression.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\Node\Expression;
+
+use Twig\Compiler;
+use Twig\Error\SyntaxError;
+use Twig\Node\Expression\Variable\ContextVariable;
+
+class ListExpression extends AbstractExpression
+{
+    /**
+     * @param array<ContextVariable> $items
+     */
+    public function __construct(array $items, int $lineno)
+    {
+        foreach ($items as $item) {
+            if (!$item instanceof ContextVariable) {
```

Comment:
Do we need to validate this here ? Isn't this also ensured by the parser when using parsing ?
For custom constructed nodes, we already have the parameter type defining the contract and we generally don't add runtime checks for array values when we have such contract.